File: ruby/hamming/hamming.rb
class Hamming

  VERSION = 1

  def initialize(strand_a, strand_b)
    fail ArgumentError unless strand_a.size == strand_b.size

    @strand_a, @strand_b = strand_a, strand_b
  end

  def self.compute(strand_a, strand_b)
    new(strand_a, strand_b).compute
  end

  def compute
    @strand_a.each_char.each_with_index.count { |nucleotide, index|
      nucleotide != @strand_b[index]
    }
  end

end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(strand_a, strand_b) {

  var count = 0;

  if (strand_a.length !== strand_b.length) {
    throw(new Error('DNA strands must be of equal length.'));
  }

  for (var i = 0; i < strand_a.length; i++) {
    if (strand_a[i] !== strand_b[i]) {
      count++
    }
  }

  return count;
}

module.exports = Hamming;
